Content/Items/Misc/SoilgunFiles/Weapons.Soilgun.cs

```diff
@@ -81,464 +89,338 @@ public override void AddRecipes()
 		}
 	}
 
-	class SoilgunGlobalItem : GlobalItem
+	class SoilgunHoldout : ModProjectile
 	{
-		public TooltipLine ammoTooltip;
+		private bool updateVelocity = true;
+
+		private bool flashed;
+		private int flashTimer;
+
+		public int ammoID;
+		public int projectileID;
+
+		public Projectile ghostProjectile = new();
+
+		public bool CanHold => Owner.channel && !Owner.CCed && !Owner.noItems;
```

Comment:
In general in the future try not to mix properties and fields like this, I generally try to do all fields then all properties